flink-libraries/flink-streaming-python/src/main/java/org/apache/flink/streaming/python/api/environment/PythonStreamExecutionEnvironment.java

```diff
@@ -0,0 +1,421 @@
+	 *
+	 * @param host
+	 * 		The host name or address of the master (JobManager), where the
+	 * 		program should be executed.
+	 * @param port
+	 * 		The port of the master (JobManager), where the program should
+	 * 		be executed.
+	 * @param config
+	 * 		The configuration used by the client that connects to the remote cluster.
+	 * @param jar_files
+	 * 		The JAR files with code that needs to be shipped to the
+	 * 		cluster. If the program uses user-defined functions,
+	 * 		user-defined input formats, or any libraries, those must be
+	 * 		provided in the JAR files.
+	 * @return A remote environment that executes the program on a cluster.
+	 *
+	 */
+	public static PythonStreamExecutionEnvironment create_remote_execution_environment(
+		String host, int port, Configuration config, String... jar_files) {
+		return new PythonStreamExecutionEnvironment(
+						StreamExecutionEnvironment.createRemoteEnvironment(host, port, config, jar_files));
+	}
+
+	/**
+	 * A thin wrapper layer over {@link StreamExecutionEnvironment#createRemoteEnvironment(
+	 * java.lang.String, int, int, java.lang.String...)}
+	 *
+	 * @param host
+	 * 		The host name or address of the master (JobManager), where the
+	 * 		program should be executed.
+	 * @param port
+	 * 		The port of the master (JobManager), where the program should
+	 * 		be executed.
+	 * @param parallelism
+	 * 		The parallelism to use during the execution.
+	 * @param jar_files
+	 * 		The JAR files with code that needs to be shipped to the
+	 * 		cluster. If the program uses user-defined functions,
+	 * 		user-defined input formats, or any libraries, those must be
+	 * 		provided in the JAR files.
+	 * @return A remote environment that executes the program on a cluster.
+	 */
+	public static PythonStreamExecutionEnvironment create_remote_execution_environment(
+		String host, int port, int parallelism, String... jar_files) {
+		return new PythonStreamExecutionEnvironment(
+						StreamExecutionEnvironment.createRemoteEnvironment(host, port, parallelism, jar_files));
+	}
+
+	private PythonStreamExecutionEnvironment(StreamExecutionEnvironment env) {
+		this.env = env;
+		this.registerJythonSerializers();
+	}
+
+	private void registerJythonSerializers() {
+		this.env.registerTypeWithKryoSerializer(PyString.class, PyObjectSerializer.class);
+		this.env.registerTypeWithKryoSerializer(PyInteger.class, PyObjectSerializer.class);
+		this.env.registerTypeWithKryoSerializer(PyLong.class, PyObjectSerializer.class);
+		this.env.registerTypeWithKryoSerializer(PyUnicode.class, PyObjectSerializer.class);
+		this.env.registerTypeWithKryoSerializer(PyTuple.class, PyObjectSerializer.class);
+		this.env.registerTypeWithKryoSerializer(PyObjectDerived.class, PyObjectSerializer.class);
+		this.env.registerTypeWithKryoSerializer(PyInstance.class, PyObjectSerializer.class);
+	}
+
+	public PythonDataStream create_python_source(SourceFunction<Object> src) throws Exception {
```

Comment:
Probably yes. All Jython user-defined functions/Classes are generated under `org.python.proxies` scope. So, using the class canonical name, we can identify these functions. WDYT?